### ISSUE TITLE
autoSyncThreshold option not respected

### ISSUE THREAD

**Issue by theodorton** [transistorsoft/react-native-background-geolocation#176]
We're using the iOS library and have some problem getting the `autoSyncThreshold` configuration to work properly. All location changes are submitted immediately, which significantly affects battery life.

## Your Environment
* Plugin version: 1.5.2
* Platform: iOS
* OS version: iOS 10.1
* Device manufacturer / model: Simulator
* Plugin config

## Expected Behavior

When configuring the plugin with the following options:

```javascript
BackgroundGeolocation.configure({
    debug: true,
    desiredAccuracy: 10,
    autoSync: true,
    autoSyncThreshold: 50,
    batchSync: true,
    maxBatchSize: 50,
    stopOnTerminate: false,
    url: `http://localhost:5000/...`,
});
```

And start tracking, I expect the plugin to wait for at least 50 location changes before POSTing anything to my URL.

## Actual Behavior

Every location change is POSTed.

## Steps to Reproduce

1. Configure with `autoSyncThreshold`.
2. Start freeway drive in simulator
3. Wait for first location change
4. HTTP request should not be sent, but is sent

## Context

Trying to reduce the amount of HTTP requests.

## Debug logs

From xcode:

```
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager locationManager:didUpdateLocations:] Enabled: 1 | isMoving: 1 | distanceFilter: 70.0
╚═══════════════════════════════════════════════════════════
- Median location accuracy: 5.0
2016-12-01 12:15:11.596 [info][tid:main][RNBackgroundGeolocation.m:379] - RCTBackgroundGeoLocation onLocationChanged
🔵-[Settings incrementOdometer:] 191175.9
✅ INSERT: D00FE1D5-59DB-4934-AD6C-153B1DD07FE2

╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush] 
╚═══════════════════════════════════════════════════════════
🔵__30-[LocationDAO allWithLocking:]_block_invoke LOCKED 1 RECORDS
🔵-[TSHttpService postBatch:] 1 records
2016-12-01 12:15:11.705 [info][tid:main][RNBackgroundGeolocation.m:458] - onHttpResponse
🔵-[TSHttpService doPost:] HTTP Response: 200
🔵-[LocationDAO destroyAll:] DESTROY: D00FE1D5-59DB-4934-AD6C-153B1DD07FE2
ℹ️__26-[LocationDAO destroyAll:]_block_invoke DESTROY SUCCESS (1)
🔵__30-[LocationDAO allWithLocking:]_block_invoke LOCKED 0 RECORDS
2016-12-01 12:15:11.706 [info][tid:main][RNBackgroundGeolocation.m:445] - onSyncComplete
```

**Comment by christocracy** [comment]
Upgrade.  This option came in 1.5.4.


https://github.com/transistorsoft/react-native-background-geolocation/blob/master/CHANGELOG.md#154---2016-11-07

**Comment by theodorton** [comment]
Looks like it was actually introduced in [1.4.0](https://github.com/transistorsoft/react-native-background-geolocation/blob/master/CHANGELOG.md#140---2016-09-21), and that the 1.5.4 change was a change:

> The plugin will ignore autoSyncThreshold when a motionchange event occurs.

Am I wrong?

**Comment by christocracy** [comment]
In `1.5.4`, I disabled `autoSyncThreshold` for `motionchange` events.

Let the city-drive continue. `autoSyncThreshold` should be respected while the plugin is in the **moving** state.  When ever the plugin changes state, from **stationary->moving**, or **moving->stationary**, `autoSyncThreshold` is ignored.

**Comment by theodorton** [comment]
This is logged as my configuration:

```
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager setConfig:] 
{
    activityRecognitionInterval = 10000;
    activityType = 1;
    autoSync = 1;
    autoSyncThreshold = 50;
    batchSync = 1;
    debug = 1;
    desiredAccuracy = 100;
    disableElasticity = 0;
    disableMotionActivityUpdates = 0;
    disableStopDetection = 0;
    distanceFilter = 10;
    enabled = 0;
    extras = 0;
    geofenceInitialTriggerEntry = 1;
    geofenceProximityRadius = 1000;
    headers = 0;
    heartbeatInterval = 60;
    isMoving = 0;
    locationAuthorizationRequest = Always;
    locationTimeout = 60;
    logLevel = 18446744073709551615;
    maxBatchSize = 50;
    maxDaysToPersist = 1;
    maxRecordsToPersist = "-1";
    method = POST;
    minimumActivityRecognitionConfidence = 70;
    params =     {
        "api_key" = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJuYWJvYmlsLWRldmVsb3BtZW50Iiwic3ViIjo1LCJhdWQiOiJuYWJvYmlsLWFwcCIsImV4cCI6MTQ4MzE3ODQ2NCwiaWF0IjoxNDgwNTg2NDYzfQ.Rsu0Uo_BcCbDSgPzhmtXxs9pIkKvdassUD1h0OXI77A";
    };
    pausesLocationUpdatesAutomatically = 0;
    preventSuspend = 0;
    schedule = 0;
    schedulerEnabled = 0;
    startOnBoot = 1;
    stationaryRadius = 25;
    stopAfterElapsedMinutes = "-1";
    stopDetectionDelay = 0;
    stopOnTerminate = 0;
    stopTimeout = 300;
    url = "http://localhost:5000/api/v1/waypoints";
    useSignificantChangesOnly = 0;
}
╚═══════════════════════════════════════════════════════════
```

This is what happens on my server, in the chrome console and in the simulator.

![des -02-2016 12-14-21](https://cloud.githubusercontent.com/assets/874365/20832145/ea96047e-b888-11e6-8d18-5d4b3b946899.gif)

Is the simulator causing this? The `activity` is always posted as `unknown` from the sim:

```json
{ ... "activity":{"type": "unknown", "confidence": 100} ... }
```

**Comment by christocracy** [comment]
I couldn't verify this yesterday.  Perhaps you have a stale copy of `TSLocationManager.framework` in your XCode project.  Which method do you use to install the plugin?
- Manual?
- RNPM?
- Cocoapod?

**Comment by christocracy** [comment]
And yes, you won't get `activity` information in Sim.  This is normal.  The sim doesn't have an accelerometer.

**Comment by christocracy** [comment]
I confirmed this issue with `1.5.2`.

This issue was fixed in [`v1.6.0`](https://github.com/transistorsoft/react-native-background-geolocation/blob/master/CHANGELOG.md#160---2016-11-18)

- [Fixed] iOS batchSync: When only 1 record in batch, iOS fails to pack the records in a JSON location: [], appending to a location: {} instead.

**Comment by theodorton** [comment]
Works after upgrading to 1.6.0 - thanks for your help!
